frontend/src/features/map_planning/components/BaseLayerConfigurator.tsx

```diff
@@ -0,0 +1,197 @@
+        if (typeof value === 'string' || Number.isNaN(value)) return;
+
+        const meters = Math.floor(realWorldLength);
+        setRealWorldLength(meters + (value / 100));
+        console.log(realWorldLength);
+    }
+
+    // Determine the scale of the image using the length of a known distance
+    const [measureState, setMeasureState]           = useState(MeasurementState.Initial);
+    const [measureLinePoints, setMeasureLinePoints] = useState<number[]>([]);
+    const [measuredLength, setMeasuredLength]       = useState(0);
+
+    const [showDistanceInputModal, setShowDistanceInputModal] = useState(false);
+
+    const onBaseStageClick = (e: KonvaEventObject<MouseEvent>) => {
+        if (e.evt.button !== 0) return;
+        
+        switch (measureState) {
+            case MeasurementState.Initial:
+                const x =  e.target.getStage()?.getRelativePointerPosition()?.x == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.x;
+                const y =  e.target.getStage()?.getRelativePointerPosition()?.y == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.y;
+                
+                setMeasureLinePoints([x ?? 0, y ?? 0]);
+                setMeasureState(MeasurementState.OnePointSelected);
+                break;
+            
+            case MeasurementState.OnePointSelected:
+                const lineLengthX = Math.abs(measureLinePoints[2] - measureLinePoints[0]);
+                const lineLengthY = Math.abs(measureLinePoints[3] - measureLinePoints[1]);
+
+                // use the pythagorean theorem to get the measured length
+                const lineLength  = Math.sqrt(lineLengthX * lineLengthX + lineLengthY * lineLengthY);
+
+                // compensate for previously applied scaling
+                setMeasuredLength(lineLength / scale * MAP_PIXELS_PER_METER);
+                
+                // Promt the user to input the real world length of the measured distance
+                setShowDistanceInputModal(true);
+
+                setMeasureState(MeasurementState.TwoPointsSelected);
+                break;
+
+            case MeasurementState.TwoPointsSelected:
+                setMeasureLinePoints([]);
+                setMeasureState(MeasurementState.Initial);
+                break;
+
+        }
+    }
+
+    const onBaseStageMouseMove = (e: KonvaEventObject<MouseEvent>) => {
+        if (measureState !== MeasurementState.OnePointSelected) return;
+
+        const x =  e.target.getStage()?.getRelativePointerPosition()?.x == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.x;
+        const y =  e.target.getStage()?.getRelativePointerPosition()?.y == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.y;
+
+        setMeasureLinePoints([measureLinePoints[0], measureLinePoints[1], x ?? 0, y ?? 0]);
+    }
+   
+    const onDistanceInputModalSubmit = () => {
+        // TODO: replace 10 with a global constant
```

Comment:
is this TODO still relevant?

Reply:
No. I have already replaced it with a constant.